src/env-inl.h

```diff
@@ -68,7 +68,10 @@ inline AsyncHooks::AsyncHooks()
     : async_ids_stack_(env()->isolate(), 16 * 2),
       fields_(env()->isolate(), kFieldsCount),
       async_id_fields_(env()->isolate(), kUidFieldsCount) {
-  v8::HandleScope handle_scope(env()->isolate());
+  v8::Isolate* isolate = env()->isolate();
+  v8::HandleScope handle_scope(isolate);
+
+  execution_async_resources_.Reset(isolate, v8::Array::New(isolate));
```

Comment:
Dry it up a bit with

```suggestion
  clear_async_id_stack();
```

? In that case the handle scope would have to be moved into the function, but it seems like that should ideally happen anyway?

Reply:
Fixed.